feathr_project/feathr/client.py

```diff
@@ -404,8 +409,33 @@ def wait_job_to_finish(self, timeout_sec: int = 300):
             return
         else:
             raise RuntimeError('Spark job failed.')
+    
+    def materialize_features(self, settings: MaterializationSettings):
+        """Materialize feature data
 
-    def materialize_features(self, feature_gen_conf_path: str = 'feature_gen_conf/feature_gen.conf'):
+        Args:
+            settings: Feature materialization settings
+        """
+        # produce materialization config
+        start_time = settings.backfill_time.start
+        end_time = settings.backfill_time.end
+        step_in_seconds = settings.backfill_time.step.total_seconds()
+        assert start_time <= end_time, "Start time {} must be earlier or equal to end time {}".format(start_time, end_time)
+        assert step_in_seconds > 0, "Step in time range should be greater than 0, but got {}".format(step_in_seconds)
+        num_delta = (settings.backfill_time.end - settings.backfill_time.start).total_seconds() / step_in_seconds
+        num_delta = math.floor(num_delta) + 1
+        for end in (end_time - timedelta(seconds=n*step_in_seconds) for n in reversed(range(num_delta))):
+            settings.backfill_time.end = end
+            config = _to_materialization_config(settings)
+            config_file_name = "feature_gen_conf/auto_gen_config_{}.conf".format(end.timestamp())
+            config_file_path = os.path.abspath(config_file_name)
+            write_to_file(content=config, full_file_name=config_file_path)
+            self._materialize_features_with_config(config_file_name)
+            self.wait_job_to_finish(timeout_sec=step_in_seconds)
+            if os.path.exists(config_file_path):
```

Comment:
I suggest using `tempfile` module for those files so that we don't have to worry about the file management

Reply:
https://docs.python.org/3/library/tempfile.html

Reply:
since we need that path under the workspace, I think it is ok to keep it as is